nanofi/tests/CTailFileTests.cpp

```diff
@@ -0,0 +1,365 @@
+}
+
+TEST_CASE("Test tail file with less than 4096 delimited chars", "[testTailFileDelimitedString]") {
+
+    const std::string delimitedString = "token1--token2--token3";
+    FileManager fm("test.txt");
+    fm.Write(delimitedString);
+    const std::string filePath = fm.getFilePath();
+    fm.CloseStream();
+
+    struct token_list tokens = tail_file(filePath.c_str(), '-', 0);
+    test_lists_equal(&tokens, std::vector<std::string>{"token1", "token2"});
+}
+
+// Although there is no delimiter within the string that is at least 4096 bytes long,
+// tail_file still creates a flow file for the first 4096 bytes.
+TEST_CASE("Test tail file having 4096 bytes without delimiter", "[testTailFile4096Chars]") {
+
+    FileManager fm("test.txt");
+    const std::string s = std::move(fm.WriteNChars(4096, 'a'));
+    const std::string filePath = fm.getFilePath();
+    fm.CloseStream();
+
+    struct token_list tokens = tail_file(filePath.c_str(), '-', 0);
+    test_lists_equal(&tokens, std::vector<std::string>{std::move(s)});
+}
+
+// Although there is no delimiter within the string that is equal to 4096 bytes or longer
+// tail_file creates a flow file for each subsequent 4096 byte chunk. It leaves the last chunk
+// if it is smaller than 4096 bytes and not delimited
+TEST_CASE("Test tail file having more than 4096 bytes without delimiter", "[testTailFileMoreThan4096Chars]") {
+
+    FileManager fm("test.txt");
+    const std::string s1 = std::move(fm.WriteNChars(4096, 'a'));
+    const std::string s2 = std::move(fm.WriteNChars(4096, 'b'));
+    const std::string s3 = "helloworld";
+    fm.Write(s3);
+    fm.CloseStream();
+
+    const uint64_t totalStringsSize = s1.size() + s2.size() + s3.size();
+    const std::string filePath = fm.getFilePath();
+    const uint64_t bytesWrittenToStream = fm.GetFileSize();
+    REQUIRE(bytesWrittenToStream == totalStringsSize);
+
+    struct token_list tokens = tail_file(filePath.c_str(), '-', 0);
+    test_lists_equal(&tokens, std::vector<std::string>{std::move(s1), std::move(s2)});
+}
+
+TEST_CASE("Test tail file having more than 4096 bytes with delimiter", "[testTailFileWithDelimitedString]") {
+
+    FileManager fm("test.txt");
+    const std::string s1 = std::move(fm.WriteNChars(4096, 'a'));
+    const std::string d1 = std::move(fm.WriteNChars(2, '-'));
+    const std::string s2 = std::move(fm.WriteNChars(4096, 'b'));
+    fm.CloseStream();
+
+    const uint64_t totalStringsSize = s1.size() + s2.size() + d1.size();
+    const std::string filePath = fm.getFilePath();
+    const uint64_t bytesWrittenToStream = fm.GetFileSize();
+    REQUIRE(bytesWrittenToStream == totalStringsSize);
+
+    struct token_list tokens = tail_file(filePath.c_str(), '-', 0);
+    test_lists_equal(&tokens, std::vector<std::string>{std::move(s1), std::move(s2)});
+}
+
+TEST_CASE("Test tail file having more than 4096 bytes with delimiter and second chunk less than 4096", "[testTailFileWithDelimitedString]") {
```

Comment:
Taking another casual look to make it look like I'm useful. I like the breadth of tests! Nice!